### ISSUE TITLE
StopIteration exception raised when narrator.explains_the_error

### ISSUE THREAD

**Issue by bandophahita** [perrygoy/screenpy#86]
Changes in 3.2.4 raise a StopIteration while the `narrator.explains_the_error()`. 
```
Traceback (most recent call last):
  File "/Users/m.wilson/projects/screenpy/screenpy/pacing.py", line 88, in wrapper
    retval = enclosed_func(*args, **kwargs)
  File "/Users/m.wilson/projects/screenpy/screenpy/actions/see.py", line 49, in perform_as
    assert_that(value, self.resolution)
  File "/Users/m.wilson/.pyenv/versions/acureg/lib/python3.7/site-packages/hamcrest/core/assert_that.py", line 58, in assert_that
    _assert_match(actual=actual, matcher=matcher, reason=reason)
  File "/Users/m.wilson/.pyenv/versions/acureg/lib/python3.7/site-packages/hamcrest/core/assert_that.py", line 73, in _assert_match
    raise AssertionError(description)
AssertionError: 
Expected: an object with length of <3>
     but: was <False>


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/m.wilson/projects/screenpy/screenpy/pacing.py", line 88, in wrapper
    retval = enclosed_func(*args, **kwargs)
  File "/Users/m.wilson/projects/py3/acutil/cpax_reg/screenplay_tests/task/is_loaded/is_loaded_page_login.py", line 30, in perform_as
    actor.should(SeeIsLoaded(*IS_LOADED_PAGE_LOGIN))
  File "/Users/m.wilson/projects/screenpy/screenpy/actor.py", line 98, in attempts_to
    self.perform(action)
  File "/Users/m.wilson/projects/screenpy/screenpy/actor.py", line 104, in perform
    action.perform_as(self)
  File "/Users/m.wilson/projects/py3/acutil/cpax_reg/screenplay_tests/task/is_loaded/is_loaded.py", line 19, in perform_as
    HasLength(len(self.tuple_of_findables))))
  File "/Users/m.wilson/projects/screenpy/screenpy/actor.py", line 98, in attempts_to
    self.perform(action)
  File "/Users/m.wilson/projects/screenpy/screenpy/actor.py", line 104, in perform
    action.perform_as(self)
  File "/Users/m.wilson/projects/screenpy/screenpy/pacing.py", line 92, in wrapper
    the_narrator.explains_the_error(exc)
  File "/Users/m.wilson/projects/screenpy/screenpy/narration/narrator.py", line 245, in explains_the_error
    adapter.error(exc)
  File "/Users/m.wilson/projects/screenpy/screenpy/narration/adapters/allure_adapter.py", line 79, in error
    p for p in plugin_manager.get_plugins() if isinstance(p, AllureListener)
StopIteration
```
I don't know much about Allure, but there but I'm guessing this exception is occurring because I don't have Allure?

Line 78 of `allure_adapter.py` seems to look for one of the plugins to be an instance of `AllureListenter`, but debugging on my end shows that the only two returned by `plugin_manager.get_plugins()` are instances of `AllureTestHelper` and `AllureTitleHelper`.  The result of this logic would produce an empty generator which `next()` barfs on.

```
        plugin = next(
            p for p in plugin_manager.get_plugins() if isinstance(p, AllureListener)
        )
```

**Comment by MarcelWilson** [comment]
My initial thought was we could just wrap the lines in AllureAdapter.error in a try/except.  But I'm not entirely convinced that's where this handling should take place.

```python
    def error(self, exc: Exception) -> None:
        try:
            plugin = next(
                p for p in plugin_manager.get_plugins() if isinstance(p, AllureListener)
            )
        except StopIteration:
            # what should happen here?
            return
        plugin.allure_logger.stop_step(
            self.step_stack[-1].uuid,
            stop=now(),
            status=get_status(exc),
            statusDetails=get_status_details(type(exc), exc, exc.__traceback__),
        )
```
This function assumes `AllureListener` is available, should the entire class make that assumption?  Should the `__init__` be doing a quick check to see if the listener is available? 

Perhaps it's enough to raise an exception at the point where `AllureAdapter()` is initialized that allure isn't available?

**Comment by MarcelWilson** [comment]
I believe we are going to run into order of events issues with trying to ensure the listener is up.  While debugging I put breakpoints in allure_pytest/plugin.py:129 and in pacing.py:17.   We instantiate `AllureAdapter()` before the allure plugin has a chance to create the listener.  It would be one thing if we always control when that is created, but if anyone is overriding the adapters in their own code (like I currently am) we couldn't control the order.  

Could we  just do something like this?

```python
    def error(self, exc: Exception) -> None:
        for p in plugin_manager.get_plugins():
            if isinstance(p, AllureListener):
                plugin = p
                break
        else:
            return
        plugin.allure_logger.stop_step(
            self.step_stack[-1].uuid,
            stop=now(),
            status=get_status(exc),
            statusDetails=get_status_details(type(exc), exc, exc.__traceback__),
        )
```

**Comment by perrygoy** [comment]
I fixed this one in 1df4cf5! I tested it locally, but wasn't sure how to enshrine it in a unit test.
